### PR TITLE
Verify Extensions with Code Starters

### PR DESCRIPTION
**Pull request by Sgitario**
Verify only the supported extensions with code starters.

### CODE REVIEW

**Comment by Sgitario** [comment]
@rsvoboda I didn't change the way the existing code start was working. Just tweaking the code similarly as done in the SpecialCharsTest in order to support also JVM and Native.

File: testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusTest.java [hunk 1]
```diff
@@ -68,18 +80,26 @@
     public static final List<List<CodeQuarkusExtensions>> supportedEx = CodeQuarkusExtensions.partition(4, CodeQuarkusExtensions.Flag.SUPPORTED);
     public static final List<List<CodeQuarkusExtensions>> notSupportedEx = CodeQuarkusExtensions.partition(1, CodeQuarkusExtensions.Flag.NOT_SUPPORTED);
     public static final List<List<CodeQuarkusExtensions>> mixedEx = CodeQuarkusExtensions.partition(1, CodeQuarkusExtensions.Flag.MIXED);
+    
+    public static final Stream<CodeQuarkusExtensions> supportedExWithCodeStarter() {
+        return Arrays.asList(CodeQuarkusExtensions.QUARKUS_CONFIG_YAML,
+                CodeQuarkusExtensions.QUARKUS_LOGGING_JSON,
+                CodeQuarkusExtensions.QUARKUS_RESTEASY,
+                CodeQuarkusExtensions.QUARKUS_RESTEASY_JACKSON,
+                CodeQuarkusExtensions.QUARKUS_SPRING_WEB).stream();
```
Comment (Sgitario) [inline testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusTest.java:89]: Let me know if you prefer loading the extensions dynamically or adding more extensions like Quarkus Qute.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -90,4 +90,4 @@ jobs:
         uses: actions/upload-artifact@v1
         with:
           name: ci-artifacts
-          path: artifacts-jvm${{ matrix.java }}.zip
\ No newline at end of file
+          path: artifacts-code-start${{ matrix.java }}.zip
```
Comment (Sgitario) [inline .github/workflows/ci.yaml:93]: The changes in `ci.yaml` are required to see the logs for each workflow action. Something we had in the windows PR, but we didn't commit.

Comment (rsvoboda) [reply .github/workflows/ci.yaml:99]: +1

File: testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusTest.java (hunk 1 above)
Comment (rsvoboda) [reply testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusTest.java:89]: Add `quarkus-qute`, it's TP and that should be covered too

File: testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusTest.java [hunk 3]
```diff
@@ -95,12 +115,46 @@ public void testRuntime(TestInfo testInfo, List<CodeQuarkusExtensions> extension
...
+            
+            LOGGER.info("Running (" + cmd + ") in directory: " + appDir);
+            pA = runCommand(cmd, appDir, runLogA);
+
             appendln(whatIDidReport, "Extensions: " + extensions.toString());
             appendln(whatIDidReport, appDir.getAbsolutePath());
-            appendlnSection(whatIDidReport, String.join(" ", devCmd));
-            pA = runCommand(devCmd, appDir, runLogA);
+            appendlnSection(whatIDidReport, String.join(" ", cmd));
+
```
Comment (rsvoboda) [inline testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusTest.java:157]: `pA = runCommand(cmd, appDir, runLogA);` from line 152 should be here, after `whatIDidReport` details are saved.

**Review by rsvoboda — CHANGES_REQUESTED** [review]
Few changes are needed

File: testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusTest.java (hunk 3 above)
Comment (Sgitario) [reply testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusTest.java:157]: Changed. In `SpecialCharsTest.java`, the `appendlnSection` methods are also done after the `runCommand` statement, should I change it here as well?

**Comment by Sgitario** [comment]
@rsvoboda I've applied all the requested changes and added Quarkus Qute extension as well.

File: testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusTest.java (hunk 3 above)
Comment (rsvoboda) [reply testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusTest.java:156]: Yes, dedicated PR please.

**Review by rsvoboda — APPROVED** [review]

File: testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusTest.java (hunk 3 above)
Comment (Sgitario) [reply testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusTest.java:156]: https://github.com/quarkus-qe/quarkus-startstop/pull/58

**Pull request merged**